### ISSUE TITLE
Improve mod information discovery

### ISSUE THREAD

**Issue by Indemnity83** [Indemnity83/Solder-Toolbelt#10]
When attempting to pack Schematica-1.7.10-1.7.6.129-universal.jar I get the following fatal error:

```
Fatal error: Cannot use object of type stdClass as array in ModCommand.php on line 222
```

Also, there is no handling for some common, but non-standard mods such as CodeChickenCore which uses a cccmod.info file (for whatever reason). Any others should be added to this thread until its closed then new issues can be opened for other one-off cases which will be evaluated for their legitimacy.

**Comment by EntranceJew** [comment]
This happens because on ModCommand.php#L199 we're trying to get the property `$modList->modlist` but the [correct name for the property](https://github.com/MinecraftForge/FML/wiki/FML-mod-information-file) is `$modList->modList` for that file.

**Comment by Indemnity83** [comment]
Yeah; I realized that after spending a few minutes in the code; I'm expanding this issue though and improving the handling of the json decode for various situations.
